tests/by-util/test_date.rs

```diff
@@ -492,3 +492,101 @@ fn test_date_empty_tz() {
         .succeeds()
         .stdout_only("UTC\n");
 }
+
+#[test]
+fn test_date_tz_utc() {
+    new_ucmd!()
+        .env("TZ", "UTC0")
+        .arg("+%Z")
+        .succeeds()
+        .stdout_only("UTC\n");
+}
```

Comment:
Also, it'd be nice to have tests that actually test that the output time is shifted properly (since that's the issue in #7498). You should pass a fixed datetime as a parameter to `date` and check that the output is exactly what you expect.

Note that adding a full test for #7498 might be tricky, as it sort of relies on the localtime not being UTC (not sure how our CI runners are configured...)

Reply:
Dear @drinkcat , I've added a test for this. as you said its kind of complicated but tried to create something general enough. 

Side note: I also did minor changes to other tests to respect the Day Light Saving changes.